File: gdscript/src/main/java/gdscript/psi/GdFuncDeclEx.java
package gdscript.psi;

import gdscript.psi.types.GdDocumented;
import org.jetbrains.annotations.*;
import com.intellij.psi.PsiElement;
import java.util.LinkedHashMap;

public interface GdFuncDeclEx extends GdExpr, GdDocumented {

  @Nullable
  GdFuncDeclIdNmi getFuncDeclIdNmi();

  @Nullable
  GdParamList getParamList();

  @Nullable
  GdReturnHint getReturnHint();

  @Nullable
  GdStmtOrSuite getStmtOrSuite();

  @NotNull
  String getInvokedReturnType();

  @Nullable
  PsiElement getReturnExpr();

  @NotNull
  LinkedHashMap<String, String> getParameters();

}

File: gdscript/src/main/java/gdscript/psi/GdFuncDeclEx.kt
package gdscript.psi

import com.intellij.psi.PsiElement
import gdscript.psi.types.GdDocumented

interface GdFuncDeclEx : GdExpr, GdDocumented {
    val funcDeclIdNmi: GdFuncDeclIdNmi?

    val paramList: GdParamList?

    val returnHint: GdReturnHint?

    val stmtOrSuite: GdStmtOrSuite?

    val invokedReturnType: String

    val returnExpr: PsiElement?

    val parameters: LinkedHashMap<String, String>
}
